analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -32,46 +32,40 @@ namespace SonarAnalyzer.Rules.CSharp
     {
-        internal const string DiagnosticId = "S2184";
+        private const string DiagnosticId = "S2184";
         private const string MessageFormat = "Cast one of the operands of this division to '{0}'.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var division = (BinaryExpressionSyntax) c.Node;
+                    var division = (BinaryExpressionSyntax)c.Node;
 
-                    if (!(c.SemanticModel.GetSymbolInfo(division).Symbol is IMethodSymbol symbol) ||
-                        symbol.ContainingType == null ||
-                        !symbol.ContainingType.IsAny(KnownType.IntegralNumbers))
+                    if (!(c.SemanticModel.GetSymbolInfo(division).Symbol is IMethodSymbol symbol)
+                        || symbol.ContainingType == null
+                        || !symbol.ContainingType.IsAny(KnownType.IntegralNumbersIncludingNative))
                     {
                         return;
                     }
 
-                    if (TryGetTypeFromAssignmentToFloatType(division, c.SemanticModel, out var assignedToType) ||
-                        TryGetTypeFromArgumentMappedToFloatType(division, c.SemanticModel, out assignedToType) ||
-                        TryGetTypeFromReturnMappedToFloatType(division, c.SemanticModel, out assignedToType))
+                    if (TryGetTypeFromAssignmentToFloatType(division, c.SemanticModel, out var assignedToType)
+                        || TryGetTypeFromArgumentMappedToFloatType(division, c.SemanticModel, out assignedToType)
+                        || TryGetTypeFromReturnMappedToFloatType(division, c.SemanticModel, out assignedToType))
                     {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(
-                            rule,
-                            division.GetLocation(),
-                            assignedToType.ToMinimalDisplayString(c.SemanticModel, division.SpanStart)));
+                        var diagnostic = Diagnostic.Create(Rule, division.GetLocation(), assignedToType.ToMinimalDisplayString(c.SemanticModel, division.SpanStart));
```

Comment:
Why create a variable if the value is used only once?

Reply:
To avoid having a long line with many method calls inside method invocations.